node/utils/getSettingsByCountry.ts

```diff
@@ -0,0 +1,21 @@
+export const getSettingsByCountry = (settings: Configuration[]) => {
+  const settingsByCountry = settings.reduce<{
+    [country: string]: CountryDataConfiguration
+  }>((groupedConfigs, configuration) => {
+    const { declarer } = configuration
+
+    const [declarerAppName] = declarer.split('@')
+
+    if (!configuration[declarerAppName]) {
+      return groupedConfigs
+    }
+
+    const countryConfig = configuration[declarerAppName]
+
+    groupedConfigs[countryConfig.countryISO] = countryConfig
```

Comment:
Shouldn't we throw an error if `groupedConfigs` does not contain the `countryConfig.countryISO` key? Or this will never happen?

Reply:
hmm could you explain more? this function should just group the configurations together by the country ISO. or you mean if there is a change the `countryConfig` doesn't have the `countryISO` property?

Reply:
> you mean if there is a chance the countryConfig doesn't have the countryISO property?

Exactly. I suppose this validation exists somewhere (if I'm not mistaken, I saw it in other PR 🤔) but I believe that the service should know how to handle this case, if it ever happens. Do you agree?

Reply:
hmm, I don't know if it is possible to happen.. because the schema marks the property as required ([see here](https://github.com/vtex-apps/country-data-settings/blob/master/configuration/schema.json#L144)), and if the configuration app isn't properly validated, you can't publish it

Reply:
I see